### PR TITLE
feat: Add container entrypoint override support to container orchestrator

### PR DESCRIPTION
**Pull request by GregoryIvo**
This PR adds the ability to change a container's entry point from the container.provider.instance in the provided metatype.

MicrosoftTeams-image

This PR also includes fixes for the following bugs:

1) broken containers would be restarted instead of replaced when changing parameters under the container.provider
2) listContainerImages would sometimes provide a null when listing images.

**Related Issue:** This PR fixes/closes {issue number}

**Description of the solution adopted:** A more detailed description of the changes made to solve/close one or more issues. If the PR is simple and easy to inderstand this section can be skipped.

**Screenshots:** If applicable, add screenshots to help explain your solution

**Any side note on the changes made:** Description of any other change that has been made, which is not directly linked to the issue resolution [e.g. Code clean up/Sonar issue resolution]

### CODE REVIEW

**Comment by GregoryIvo** [comment]
@marcellorinaldo This PR should be ready to go. Kindly review it at your earliest convenience. Cheers!

File: kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ContainerConfiguration.java [hunk 1]
```diff
@@ -200,6 +201,17 @@ public int getImageDownloadTimeoutSeconds() {
         return this.imageConfig.getimageDownloadTimeoutSeconds();
     }
 
+    /**
+     * Returns a List<String> of container entry points. If this field is left empty
+     * the default container entrypoint will be used.
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ContainerConfiguration.java]: I wouldn't explain in the javadocs what the code will do. Maybe just say that an empty list can be returned if no entry points are specified

File: kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstanceOptions.java [hunk 2]
```diff
@@ -149,19 +152,19 @@ private List<String> parseEnvVars(String containerVolumeString) {
...
-        if (containerDevice.isEmpty()) {
-            return deviceList;
+        if (stringToSplit.isEmpty()) {
+            return stringList;
         }
 
-        for (String entry : containerDevice.trim().split(",")) {
-            deviceList.add(entry.trim());
+        for (String entry : stringToSplit.trim().split(",")) {
+            stringList.add(entry.trim());
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.container.provider/src/main/java/org/eclipse/kura/container/provider/ContainerInstanceOptions.java]: I would add a check before adding to `stringList`:
`entry.trim().length() > 0`
To prevents having lists like `[,,xyz]`

File: kura/test/org.eclipse.kura.container.provider.test/src/test/java/org/eclipse/kura/container/provider/ContainerInstanceOptionsTest.java [hunk 3]
```diff
@@ -639,6 +642,7 @@ private void givenDefaultProperties() {
         this.properties.put(REGISTRY_USERNAME, DEFAULT_REGISTRY_USERNAME);
         this.properties.put(REGISTRY_PASSWORD, DEFAULT_REGISTRY_PASSWORD);
         this.properties.put(IMAGES_DOWNLOAD_TIMEOUT, DEFAULT_IMAGES_DOWNLOAD_TIMEOUT);
+        this.properties.put(CONTAINER_ENTRY_POINT, DEFAULT_CONTAINER_ENTRY_POINT);
```
Comment (marcellorinaldo) [inline kura/test/org.eclipse.kura.container.provider.test/src/test/java/org/eclipse/kura/container/provider/ContainerInstanceOptionsTest.java:653]: I would add some test cases for the new property. Corner cases like given ",,," or "xyz,,,".

**Review by marcellorinaldo — CHANGES_REQUESTED** [review]
Tested working correctly on raspberry pi 4.

**Comment by GregoryIvo** [comment]
@marcellorinaldo  I made your suggested changes, kindly re-review at your earliest convenience. :)

**Review by marcellorinaldo — APPROVED** [review]

**Review by marcellorinaldo — APPROVED** [review]
